stream_alert_cli/manage_lambda/version.py

```diff
@@ -32,15 +32,15 @@ class LambdaVersion(object):
     All StreamAlert setups should start with "$LATEST".
     """
 
-    def __init__(self, **kwargs):
+    def __init__(self, config, package):
         """Initialize the version publishing
 
         Keyword Args:
             config (CLIConfig): Loaded StreamAlert CLI Config
             package (LambdaPackage): The created Lambda Package
         """
-        self.config = kwargs['config']
-        self.package = kwargs['package']
+        self.config = config
+        self.package = package
```

Comment:
👍 😸

Reply:
we've talked about this before but:

1) kwargs adds weird complexity like having to do a dict lookup inside of the called function. this is especially the case when there are only a few args or the function doesn't have a bunch of varied arguments it should expect
2) the caller can still refer to the argument by name, (without using kwargs) so the interface doesn't change.
3) not using kwargs has the added advantage of ensuring a function is actually used properly... usage of kwargs exclusively would allow you to call the function with no args at all which would fail in most areas use it throughout code. this is bad design and the use would never know what they were doing wrong
4) here is a good use of kwargs (ignore docstring that I have to update): https://github.com/airbnb/streamalert/pull/679/files#diff-e92d6322b319cf27f8967bef12e023d4R51
5) bad use of kwargs: https://github.com/airbnb/streamalert/pull/679/files#diff-e92d6322b319cf27f8967bef12e023d4L410